contracts/gateway/RedeemPool.sol

```diff
@@ -0,0 +1,447 @@
+            _amount,
+            _beneficiary,
+            _gasPrice,
+            _gasLimit,
+            _nonce,
+            _hashLock,
+            _cogateway
+        );
+
+    }
+
+    /**
+     * @notice It can only be called by redeemer of the redeemer request.
+     *
+     * @param _amount Amount that is to be redeem.
+     * @param _beneficiary The address in the origin chain where value token
+     *                     will be returned.
+     * @param _gasPrice Gas price that redeemer is ready to pay to get the redeem
+     *                  and unstake process done.
+     * @param _gasLimit Gas limit that redeemer is ready to pay.
+     * @param _nonce Redeem proxy nonce specific to co-gateway.
+     * @param _cogateway Address of the cogateway contract.
+     */
+    function revokeRedeemRequest(
+        uint256 _amount,
+        address _beneficiary,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        uint256 _nonce,
+        EIP20CoGatewayInterface _cogateway
+    )
+        external
+    {
+        address redeemer = msg.sender;
+        bytes32 redeemRequestHash = hashRedeemRequest(
+            _amount,
+            _beneficiary,
+            _gasPrice,
+            _gasLimit,
+            _nonce,
+            redeemer,
+            address(_cogateway)
+        );
+
+        require(
+            redeemRequestHash == redeemRequestHashes[redeemer][address(_cogateway)],
+            'Redeem request must exists.'
+        );
+
+        removeRedeemRequest(redeemer, _amount, _cogateway);
+        emit RedeemRevoked(redeemer, redeemRequestHash);
+    }
+
+    /**
+     * @notice It can only be called by Facilitator/Worker. It rejects the redeem request.
+     *
+     * @param _amount Amount that is to be redeemed.
+     * @param _beneficiary The address in the origin chain where value token
+     *                     will be returned.
+     * @param _gasPrice Gas price that redeemer is ready to pay to get the redeem
+     *                  and unstake process done.
+     * @param _gasLimit Gas limit that redeemer is ready to pay.
+     * @param _nonce Redeem proxy nonce specific to co-gateway.
+     * @param _redeemer RedeemerProxy contract address of redeemer.
+     * @param _cogateway Address of the cogateway.
+     */
+    function rejectRedeemRequest(
+        uint256 _amount,
+        address _beneficiary,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        uint256 _nonce,
+        address _redeemer,
+        EIP20CoGatewayInterface _cogateway
+    )
+        external
+        onlyWorker
+    {
+        bytes32 redeemRequestHash = hashRedeemRequest(
+            _amount,
+            _beneficiary,
+            _gasPrice,
+            _gasLimit,
+            _nonce,
+            _redeemer,
+            address(_cogateway)
+        );
+
+        require(
+            redeemRequestHash == redeemRequestHashes[_redeemer][address(_cogateway)],
+            'Redeem request must exists.'
+        );
```

Comment:
Should this `require` and at line 299 be placed in `removeRedeemRequest` function?
What do you think?